lib/src/multi_sig/multi_sig_builder.ts

```diff
@@ -0,0 +1,329 @@
+            coSignerPublicKey,
+            coSignerNonce,
+        );
+        this.coSignerNonces!.push(coSignerNonce);
+
+        return this.session;
+    }
+
+    /**
+     * generate partial signature, noted that it can only be taken after collected all nonces
+     * @returns {partialSignature}
+     * @memberof MultiSigBuilder
+     */
+    public partialSign(): Buffer | undefined {
+        if (!this.hasColletedAllNonces())
+            throw new Error(
+                "Session can be partially signed only after all co-signer's nonces are added",
+            );
+
+        const {
+            partialSignature,
+            incompleteSession,
+        } = native.multiSig.partialSign(this.session);
+
+        this.partialSignature = partialSignature;
+        this.session = incompleteSession;
+
+        return this.partialSignature;
+    }
+
+    /**
+     * add co-signer's partial signature to current session
+     * @param coSignerPublicKey external co-signer's publickey
+     * @param coSignerSignature external co-signer's partial signature
+     * @returns {session} session after add other nonce commitment
+     * @memberof MultiSigBuilder
+     */
+    public addPartialSignature(
+        coSignerPublicKey: Buffer,
+        coSignerSignature: Buffer,
+    ): string | undefined {
+        ow(coSignerPublicKey, ow.buffer);
+        ow(coSignerSignature, ow.buffer);
+
+        if (this.hasThisSignerSignature(coSignerSignature))
+            throw new Error(
+                "This co-signer's signature has already been added",
+            );
+
+        this.session = native.multiSig.addPartialSignature(
+            this.session,
+            coSignerPublicKey,
+            coSignerSignature,
+        );
+        this.coSignerSignatures!.push(coSignerSignature);
+
+        return this.session;
+    }
+
+    /**
+     * generate final signature, noted that it can only be taken after collected all partial signatures
+     * @returns {signature}
+     * @memberof MultiSigBuilder
+     */
+    public sign(): string | undefined {
+        if (!this.hasColletedAllPartialSignatures())
+            throw new Error(
+                "Session can be final signed only after all co-signer's partial signatures are added",
+            );
+
+        this.signature = native.multiSig.sign(this.session);
+
+        return this.signature;
+    }
+
+    /**
+     * verify the final signature valid or not
+     * @param signature self or other's signature
+     * @returns {boolean}
+     * @memberof MultiSigBuilder
+     */
+    public verify(_signature?: string): boolean {
+        let signature = _signature;
+        if (!_signature) {
+            if (!this.signature)
+                throw new Error('Own signature is now ready for verification');
+            signature = this.signature;
+        }
+
+        return native.multiSig.verify(
+            signature,
+            this.message,
+            this.signerPublicKeys,
+        );
+    }
+
```

Comment:
Thumbs up for defining methods for conveying meanings and increasing readability.